server.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+const net = require('net');
+const server = net.createServer();
+const parseMessage = require('./lib/parse-message.js');
+
+let sockets = [];
+
+server.on('connection', function(socket){
+  console.log('a client connected');
+  socket.write('welcome to wack chat');
+  socket.username = `user_${Math.floor(Math.random()*100)}`;
+  sockets.push(socket);
+
+  socket.on('data', function(buffer){
+    console.log(buffer.toString());
+    let message = buffer.toString();
+    if(message.startsWith('/nick'))
+      return parseMessage.nickCommand(message, socket);
+
+    if(message.startsWith('/dm'))
+      return parseMessage.dmCommand(message, sockets);
+
+    if(message.startsWith('/users'))
+      return parseMessage.users(sockets, socket);
+
+    if(message.startsWith('/troll'))
+      return parseMessage.trolololol(message, sockets);
+
+    if(message.startsWith('/ban')) {
+      return parseMessage.ban(message, sockets);
+
+    }
+
+    sockets.forEach(s => {
```

Comment:
Ha, forget what I said about the ``all`` command, that was just something Duncan had me do for my last lab so I forgot that was not an FT for this lab. Nevertheless, still a good idea to take this ``forEach`` method out from a UX standpoint.